test/e2e/service_externalip_test.go

```diff
@@ -779,6 +779,30 @@ func testExternalIPAccess(t *testing.T, data *TestData) {
 					externalIP, host, err := waitExternalIPConfigured(service)
 					require.NoError(t, err)
 
+					featureGate, err := GetAgentFeatures()
+					require.NoError(t, err)
+					// A route to the client IP is required for DSR mode.
+					if featureGate.Enabled(features.LoadBalancerModeDSR) {
+						nodeInfo := getNodeByName(host)
```

Comment:
technically you should check the `defaultLoadBalancerMode` in the Agent Config as well, I think it would be better

Could you add a helper function `isDefaultLoadBalancerModeDSR` (or maybe `getDefaultLoadBalancerMode`), which will look both at the feature gate and at `antreaProxy.defaultLoadBalancerMode`?

Reply:
added. thx!